controller/linodecluster_controller.go

```diff
@@ -117,24 +117,48 @@ func (r *LinodeClusterReconciler) reconcile(
 	// Always close the scope when exiting this function so we can persist any LinodeCluster changes.
 	defer func() {
 		// Filter out any IsNotFound message since client.IgnoreNotFound does not handle aggregate errors
-		if err := clusterScope.Close(ctx); utilerrors.FilterOut(err, apierrors.IsNotFound) != nil && reterr == nil {
+		// if len(clusterScope.LinodeCluster.Finalizers) == 0 {
+		// 	return
+		// }
+		if err := clusterScope.Close(ctx); utilerrors.FilterOut(util.UnwrapError(err), apierrors.IsNotFound) != nil && reterr == nil {
 			logger.Error(err, "failed to patch LinodeCluster")
 			reterr = err
 		}
 	}()
 
 	// Handle deleted clusters
 	if !clusterScope.LinodeCluster.DeletionTimestamp.IsZero() {
-		return res, r.reconcileDelete(ctx, logger, clusterScope)
+		err := r.reconcileDelete(ctx, logger, clusterScope)
+		if err != nil {
+			if !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+				logger.Info("re-queuing cluster/nb deletion")
+
+				res = ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}
+
+				return res, nil
+			}
+			return res, err
+		}
+		return res, nil
 	}
 
-	if err := clusterScope.AddFinalizer(ctx); err != nil {
+	err := clusterScope.AddFinalizer(ctx)
+	if err != nil {
+		logger.Error(err, "failed to update cluster finalizer")
 		return res, err
 	}
 
 	// Create
 	if clusterScope.LinodeCluster.Spec.ControlPlaneEndpoint.Host == "" {
-		if err := r.reconcileCreate(ctx, logger, clusterScope); err != nil {
+		err := r.reconcileCreate(ctx, logger, clusterScope)
+		if err != nil {
+			if !reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
+				logger.Info("re-queuing cluster/nb creation")
+
+				res = ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}
+
+				return res, nil
+			}
```

Comment:
should be able to condense this to:
```suggestion
		if err := r.reconcileCreate(ctx, logger, clusterScope); err != nil &&
			!reconciler.HasConditionSeverity(clusterScope.LinodeCluster, clusterv1.ReadyCondition, clusterv1.ConditionSeverityError) {
				logger.Info("re-queuing cluster/nb creation")

				return ctrl.Result{RequeueAfter: reconciler.DefaultClusterControllerReconcileDelay}, nil
			}
```